pkg/authorization/conditions/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package conditions
+
+import (
+	"net/http"
+
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+)
+
+// Condition is the underlying building block of a Rule
+type Condition interface {
+	// Match returns a bool based on details of the Request or Session
+	Match(*http.Request, *sessions.SessionState) bool
+}
```

Comment:
Tempted to say we should leave it as just matching on the request `Match(*http.Request) bool` until we start using this to validate sessions. At present it is always nil

Reply:
That's not unreasonable.  It shouldn't be too much of a pain to change the api later, no?